### PR TITLE
Added support for Byte

### PR DESCRIPTION
**Pull request by ajaychandran**

### CODE REVIEW

**Review by leifwickland — COMMENTED** [review]
@ajaychandran Thanks for the contribution!

Unfortunately CI is failing currently because one of the deps can't be downloaded. It builds for me locally.

Please add units tests for this new functionality in `tests/src/test/scala/pureconfig/BasicConvertersSuite.scala` via `checkArbitrary`.

Please add a `byteConfigWriter` in `core/src/main/scala/pureconfig/BasicWriters.scala`.

Please add `Byte` to line 13 of `docs/src/main/tut/docs/built-in-supported-types.md`.

**Comment by ajaychandran** [comment]
@leifwickland Thanks for the review!

The writer instance was added in the first commit.

File: tests/src/test/scala/pureconfig/BasicConvertersSuite.scala [hunk 1]
```diff
@@ -94,6 +94,8 @@ class BasicConvertersSuite extends BaseSuite {
     ConfigValueFactory.fromAnyRef("no") -> false,
     ConfigValueFactory.fromAnyRef("off") -> false)
 
+  checkArbitrary[Byte]
```
Comment (leifwickland) [inline tests/src/test/scala/pureconfig/BasicConvertersSuite.scala]: Nitpick: Please move this down to around line 120 near where `Short` is checked.

File: core/src/main/scala/pureconfig/BasicReaders.scala [hunk 2]
```diff
@@ -33,6 +33,8 @@ trait PrimitiveReaders {
 
   implicit val booleanConfigReader = ConfigReader.fromCursor(_.asBoolean)
 
+  implicit val byteConfigReader = ConfigReader.fromCursor(_.asByte)
```
Comment (leifwickland) [inline core/src/main/scala/pureconfig/BasicReaders.scala]: Nitpick: Please move this down to around 60 after the reader for `Short`.

File: core/src/main/scala/pureconfig/BasicWriters.scala [hunk 3]
```diff
@@ -23,6 +23,7 @@ trait PrimitiveWriters {
   implicit val stringConfigWriter = ConfigWriter.forPrimitive[String]
   implicit val charConfigWriter = ConfigWriter.toDefaultString[Char]
   implicit val booleanConfigWriter = ConfigWriter.forPrimitive[Boolean]
+  implicit val byteConfigWriter = ConfigWriter.forPrimitive[Byte]
```
Comment (leifwickland) [inline core/src/main/scala/pureconfig/BasicWriters.scala]: Nitpick: Please move this down to around 30 after the writer for `Short`.

File: docs/docs/docs/built-in-supported-types.md [hunk 4]
```diff
@@ -10,7 +10,7 @@ using those types, users don't have to provide anything else in order to be able
 
 The currently supported basic types are:
 
-- `String`, `Boolean`, `Double` (standard and percentage format ending with `%`), `Float` (also supporting percentage),
+- `String`, `Boolean`, `Byte`, `Double` (standard and percentage format ending with `%`), `Float` (also supporting percentage),
```
Comment (leifwickland) [inline docs/docs/docs/built-in-supported-types.md]: Nitpick: Please put `Byte` after `Short` with the rest of the integer types.

**Review by leifwickland — APPROVED** [review]
@ajaychandran Thanks for fixing my nitpicks. This looks good to me. 

Let's wait for one of the other maintainers to review it before we merge.

@ruippeixotog @jcazevedo Can you please remind me what we need to do when we change the tut docs in order to publish the updated pages?

**Review by jcazevedo — APPROVED** [review]
@ajaychandran: Thanks for your PR!

> Can you please remind me what we need to do when we change the tut docs in order to publish the updated pages?

We run the `scripts/update_website.sh` script to update the website. However, we only do it after a release, so that https://pureconfig.github.io/ is in sync with the latest published version.

**Pull request merged**